hw/top_earlgrey/syn/asic.constraints.sdc

```diff
@@ -181,6 +191,24 @@ set_output_delay ${SPI_DEV_OUT_DEL} [get_ports SPI_DEV_D1]   -clock SPI_DEV_CLK
 set_output_delay ${SPI_DEV_OUT_DEL} [get_ports SPI_DEV_D2]   -clock SPI_DEV_CLK
 set_output_delay ${SPI_DEV_OUT_DEL} [get_ports SPI_DEV_D3]   -clock SPI_DEV_CLK
 
+# False path from CSb to return to host as CSb for that path behaves as reset.
+set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+               -through [get_pins top_earlgrey/u_spi_device/cio_sd_en_o*]
+#set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+#               -through [get_pins top_earlgrey/u_spi_device/cio_sd_en_o[1]]
+#set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+#               -through [get_pins top_earlgrey/u_spi_device/cio_sd_en_o[2]]
+#set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+#               -through [get_pins top_earlgrey/u_spi_device/cio_sd_en_o[3]]
+set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+               -through [get_pins top_earlgrey/u_spi_device/cio_sd_o*]
+#set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+#               -through [get_pins top_earlgrey/u_spi_device/cio_sd_o[1]]
+#set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+#               -through [get_pins top_earlgrey/u_spi_device/cio_sd_o[2]]
+#set_false_path -through [get_pins top_earlgrey/u_spi_device/cio_csb_i] \
+#               -through [get_pins top_earlgrey/u_spi_device/cio_sd_o[3]]
```

Comment:
I'm not an expert on this part, but would it be better to have a max delay vs false path? 
I think most likely it won't matter, but we probably don't want a case where the reset could effectively take super long to propagate right?

Reply:
I am not an expert either (maybe newbie?) , CSb and SPI_CLK are quite special. Without false_path, the path violates no matter how the option I added. The path starts from pinmux and back to pinmux, which are assumed as legitimate datapath, not the reset path.

Maybe I should try `set_ideal_network` for scan muxed csb? ... let me try that.

Reply:
I used to be a constraints hacker / STA guy for many years:

-I had a strong preference for avoiding set_false_path where possible.  max_delay or ideal_network are better (I think you can put a latency on the ideal network to model the insertion delay at RTL before the buffer chains are there as well, if you need it to align hold times / sample windows)

-I had an even stronger preference for avoiding "naked" -through on any of these, because I had bad luck with -through overexpanding to hit too many start/endpoints.  I liked -from with -through, or -to with -through, (or just -from with -to if that covers the correct/full cone, but sometimes you need -through added to limit to a subcone), since having at least a -from or -to will limit the scope that the false_path will cover.  (Basically my nightmare scenario as an STA engineer was an overzealous false_path that masked real timing paths that were violating.)

-The final form of this should usually be a tcl for loop over each bit, for constraint maintenance reasons.  No one did this, thankfully, but you absolutely do *not* want to use cio_sd_o[*] or something like that.  (For one, that can hide undiagnosed and erroneous paths that cross between the bit slices.)

Reply:
Hum, then should be `set_ideal_network` but I can't do that for the csb reset as it passes `prim_clock_mux2` module. That module's detail impl is process dependant. DC requires ideal network to be pin of leaf cell or port. We can't make CSb port to be ideal network as it can be also data path (passthrough).

max_delay won't help to avoid timing violation on this path too. 

Anyway, I think as of now, false_path is the only option, unless we want to expose process variant here.

Reply:
@eunchan 
can you file an issue to track this one (for the ideal network, I do agree with you we need to do that one)? 
We should probably handle it as a separate PR similar to what we do for the driving / load pins on clocks.
https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/syn/asic.constraints.sdc#L74

Regarding the path though, is this not a realistic path? 

CSb (from host) -> reset internal flop -> MISO pin? 

I thought for example there would be such a case, and that's why I was thinking we need some kind of large max delay to ensure the last leg propagate wouldn't be too long for whatever reason.

Reply:
OK I will create an issue for this. Adding size only buffer to the passthrough path may not be a solution as it introduce more delay. 

For the data path. It exists, but not happens in the normal datapath. CSb de-assertion is assumed as reset, so any delay on MISO lines shouldn't be a problem.

Reply:
Ah.. maybe it's OK. As CSb to be reset, doesn't  do anything on passthrough.